src/compose.ts

```diff
@@ -158,30 +162,39 @@ function normalizeObjectToComposition(
 			c = { version: SchemaVersion.v2_0, services: c };
 		// FIXME: perform attribute migration
 		case SchemaVersion.v2_0:
-			c.version = SchemaVersion.v2_1;
+			c.version = SchemaVersion.v2_4;
```

Comment:
You can't do that -- the idea is that in each `case` we're incrementally doing any required transformations in order to normalize the composition to the internal working representation. Unless you're migrating `2.1` input to `2.4` you shouldn't need to change this.